### PR TITLE
Allow admin to edit site announcement using admin tools

### LINKED ISSUE #2010: Make site banner announcement configurable and update with wise4 phaseout info
1. Allow wise admins to set the site's banner announcement content via `/admin/portal/manage`.
2. Update banner text and contents with wise4 phaseout info and date (Oct 1).
3. Update warning text for wise4 units in the site library and run listings (e.g. "Warning! Legacy WISE units will be deprecated on Oct 1, 2019. Please find an alternate unit to use in the future. If you want to continue using this unit, contact us...")

### PR DESCRIPTION
**Pull request by breity**
Admins can now set the site announcement banner and content via `/admin/portal/manage`. I also updated the legacy unit warning messages to reflect pending deprecation.

To test:
- Add 'announcement' column to 'portal' table in local db and add default data:
```
ALTER TABLE portal
ADD announcement MEDIUMTEXT;
UPDATE portal SET announcement = '{"visible": false,"bannerText": "","bannerButton": "","title": "","content": "","buttons": []}' WHERE id = 1;
```
- Load site homepage and make sure no announcement banner appears.
- Log in as an admin and navigate to `/admin/portal/manage`.
- Edit the announcement data and save. For example:
```
{
	"visible": true,
	"bannerText": "Legacy unit support is ending.",
	"bannerButton": "Learn More",
	"title": "Ending support for legacy units",
	"content": "<p>On October 1, 2019, we will be removing support for legacy WISE units. While WISE users will still be able to view legacy unit content in preview mode and review student work from previous classroom implementations, they will no longer be able to create new legacy units or run legacy units with their students.</p>",
	"buttons": [{
		"text": "Contact Us",
		"link": "https://wise.berkeley.edu/contact",
                "color": "accent"
	}]
}
```
- Navigate to site home and make sure the announcement shows up and that clicking the "Learn More" button opens the announcement details dialog.
- Create a new run using a legacy (wise4) unit.
- Make sure the text "(Legacy Unit)" appears next to the new run's title in the teacher "Class Schedule".

Resolves #2010.

### CODE REVIEW

File: src/main/java/org/wise/portal/presentation/web/controllers/admin/ManagePortalController.java [hunk 1]
```diff
@@ -93,7 +94,11 @@ protected void savePortalChanges(
         portalService.updatePortal(portal);
         addOfficialTagToProjectLibraryGroup(val);
         modelMap.put("msg", "success");
-      } else {
+      } else if (attr.equals("announcement")) {
+        portal.setAnnouncement(val);
+        portalService.updatePortal(portal);
+        modelMap.put("msg", "success");
+      }else {
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/admin/ManagePortalController.java]:
```suggestion
      } else {
```

File: src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java [hunk 2]
```diff
@@ -23,23 +23,41 @@
  */
 package org.wise.portal.presentation.web.controllers.admin;
 
+import org.json.JSONException;
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java]: remove import

File: src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java [hunk 3]
```diff
@@ -23,23 +23,41 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java]: remove import

File: src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java [hunk 4]
```diff
@@ -23,23 +23,41 @@
...
+
   @GetMapping(value = {"", "/student", "/student/**", "/teacher", "/teacher/**", "/login", 
       "/login/**", "/join", "/join/**", "/news", "/about", "/features", "/privacy", "/news",
       "/contact", "/contact/**", "/help", "/help/**", "/forgot", "/forgot/**"})
   protected String showSite() {
     return "forward:/index.html";
   }
+
+  @ResponseBody
+  @GetMapping(value = "/announcement")
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java]:
```suggestion
  @GetMapping("/announcement")
```

File: src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java [hunk 5]
```diff
@@ -23,23 +23,41 @@
...
+  protected String getAnnouncement(ModelMap modelMap) throws JSONException, ObjectNotFoundException {
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/presentation/web/controllers/admin/SiteController.java]:
```suggestion
  protected String getAnnouncement() throws ObjectNotFoundException {
```

**Comment by breity** [comment]
Great, I updated with your code changes @hirokiterashima.

**Comment by hirokiterashima** [comment]
Looks good!

**Pull request merged**